_posts/2024-10-26-a-faster-liquidsoap-part-deux.md

```diff
@@ -9,281 +9,310 @@ _The memory part_
+We [explored these issues in a previous post](../2023-07-09-memory-management) and gave an example of parameters that
+govern this trade-off. In the future, we would like to explore alternative memory allocation strategies to better
+control short-term memory allocations inside the application.
 
 ### How to report memory usage
 
-Now that we've seen different elements of memory usage, how can we report memory usage? It turns out that this is tricky and OS-dependent!
+Now that we've seen different elements of memory usage, how can we report memory usage? It turns out that this is tricky
+and OS-dependent!
 
-In linux, if you use `top` or similar tool, the most common measure of memory consumption is `RSS` for [_Resident set size_](https://en.wikipedia.org/wiki/Resident_set_size)
+In Linux, if you use `top` or a similar tool, the most common measure of memory consumption is `RSS` for
+[_Resident set size_](https://en.wikipedia.org/wiki/Resident_set_size).
 
-However, this number does include shared memory so, again, 10 processes using `30MB` of RSS memory does not mean that a total
-of `300MB` of the system's memory is used.
+While this number includes shared memory, it does not imply that if `10` processes each use `30MB` of RSS memory, the
+total memory used by the system is `300MB`.
 
-Also, if a program, like liquidsoap, is using a large amount of shared libraries,
-the total amount of reported `RSS` memory will be large but unrelated to the memory effectively used by the application!
+Moreover, if a program like Liquidsoap depends on many shared libraries, the total reported `RSS` memory may be quite
+large; however, it might not accurately reflect the actual memory the application occupies.
 
-Likewise, in [the previously mentioned issue](https://github.com/savonet/liquidsoap/issues/4058), it took us a while to realize that, the tool that
-our user was using to report the memory usage of the process from their virtualized environment was also including the page cache, leading to
-the wrong observations regarding the memory effectively allocated by the application itself.
+Likewise, in [the previously mentioned issue](https://github.com/savonet/liquidsoap/issues/4058), it took us a while to
+realize that the tool our user was using to report the process's memory usage from their virtualized environment was
+also including the page cache, leading to the wrong observations regarding the memory effectively allocated by the
+application itself.
 
-In the [ocaml-mem_usage](https://github.com/savonet/ocaml-mem_usage) module, we implemented several OS-specific methods to measure the memory used by the application
-and separate shared memory from private memory.
+In the [ocaml-mem_usage](https://github.com/savonet/ocaml-mem_usage) module, we implemented several OS-specific methods
+to measure the application's memory use and separate shared memory from private memory.
```

Comment:
I'm not sure that this is more clear.

Reply:
This fragment is literally the same, except the word order.
```diff
- the memory used by the application
+ the application's memory use
```

Reply:
Thank you, I have undone this change.

Reply:
Thanks. Yes, it is the same meaning but I think that, for technical documentation being explicit at the expense of being repetitive/less idiomatic is always better. So, "the memory used by the application" is always more clear, especially for non-native reader, than "the application's memory" which requires the understanding of the english genitive, a construct specific to that language.